### PR TITLE
refactor(pages): rework airdrops create page

### LINKED ISSUE #57: Update create airdrop page
https://www.figma.com/file/FHGAAxlpvIkCo8qoYlzqqA/?node-id=173%3A3038

image

### PR DESCRIPTION
**Pull request by grikomsn**
Fixes #57

## Description

This week-long PR updates the create airdrop page according to the Figma spec.

## Changes

- [x] match create airdrop page designs with figma
- [x] extract form control into separate component
- [x] ~~rework form logic to react-hook-form~~ (separate PR)
- [x] ~~exhaustive form component splitting~~ (separate PR)

## Screenshots

https://cdn.discordapp.com/attachments/933515307597848656/947554886529196072/junotools-create-airdrop-page-2022-02-28.gif

## Testing Steps

As a reviewer, what steps should I take to verify this is working correctly?

- ensure all inputs are fillable and clickable
- try both click to upload csv and drag file to input area

## Links

\-

### CODE REVIEW

File: pages/airdrops/create.tsx [hunk 1]
```diff
@@ -289,209 +321,235 @@ const CreateAirdrop: NextPage = () => {
...
+          title="CW20 Address"
+          subtitle=" Address of the CW20 token that will be airdropped."
+          htmlId="airdrop-cw20"
+        >
+          <Input
+            id="airdrop-cw20"
+            name="cw20"
+            placeholder="juno1234567890abcdefghijklmnopqrstuvwxyz..."
             value={cw20TokenAddress}
             onChange={(e) => setCW20TokenAddress(e.target.value)}
```
Comment (XanderJL) [inline pages/airdrops/create.tsx:362]: Regardless of library used, I'm curious if anyone has a plan for a validation rule to check if the string is a valid wallet address 👀

File: pages/airdrops/create.tsx [hunk 2]
```diff
@@ -289,209 +321,235 @@ const CreateAirdrop: NextPage = () => {
...
-          'py-2 px-4 w-full font-bold',
-          'bg-plumbus-60 hover:bg-plumbus-50 rounded focus:ring',
-          { 'opacity-50 cursor-wait': loading }
-        )}
-        disabled={loading}
-        onClick={uploadJSONOnClick}
+      <div
+        className={clsx('flex justify-end pb-6', {
+          'sticky right-0 bottom-0': isValidToCreate,
+        })}
```
Comment (XanderJL) [inline pages/airdrops/create.tsx:499]: I don't see anything wrong with the logic, but breaking this form up into section components would be very helpful for readability. As it stands, the file is >500 lines long

**Review by XanderJL — CHANGES_REQUESTED** [review]
I think that `create.tsx` file should definitely be broken up to improve readability/comprehension, but other than that, this looks very solid!

File: pages/airdrops/create.tsx [hunk 3]
```diff
@@ -24,16 +25,45 @@ import {
...
+    id: 'height',
+    title: 'Block Height',
+    subtitle: 'Choose a specific block height for this airdrop to begin.',
+  },
+  {
+    id: 'timestamp',
+    title: 'Timestamp',
+    subtitle: 'Specific a calendar date and time of day.',
+  },
+]
```
[the remark this answers is not in the document]
Comment (grikomsn) [reply pages/airdrops/create.tsx]: Can't figure out where to put it until we also split all business logics 🤔

File: pages/airdrops/create.tsx [hunk 4]
```diff
@@ -53,10 +83,10 @@ const CreateAirdrop: NextPage = () => {
   const [cw20TokenAddress, setCW20TokenAddress] = useState('')
   const [start, setStart] = useState('')
   const [startDate, setStartDate] = useState<Date | null>(null)
-  const [startType, setStartType] = useState<StartEndValue>('height')
+  const [startType, setStartType] = useState<StartEndValue>('null')
   const [expiration, setExpiration] = useState('')
   const [expirationDate, setExpirationDate] = useState<Date | null>(null)
-  const [expirationType, setExpirationType] = useState<StartEndValue>('height')
+  const [expirationType, setExpirationType] = useState<StartEndValue>('null')
```
[the remark this answers is not in the document]
Comment (grikomsn) [reply pages/airdrops/create.tsx:90]: That is in the works. Will attempt refactoring the whole business logics to rhf.

File: pages/airdrops/create.tsx (hunk 1 above)
Comment (grikomsn) [reply pages/airdrops/create.tsx:362]: Probably in rhf we can add a custom validator whether the length is correct and it has a `juno` prefix. Don't know if this should be in this PR scope or not.

File: pages/airdrops/create.tsx (hunk 2 above)
Comment (grikomsn) [reply pages/airdrops/create.tsx:499]: If we split these, prop drilling will be required to pass all form states and handlers. Again, we can iterate further more until we split all business logics.

**Comment by grikomsn** [comment]
@XanderJL coincidentally I pushed several commits when your review came 😅

**Review by XanderJL — APPROVED** [review]
LGTM 🚀

**Pull request merged**